src/main/scala/lms/core/codegen.scala

```diff
@@ -116,3 +128,161 @@ trait ExtendedCodeGen {
   // Head Function: emitAll take a graph and generate the code.
   def emitAll(g: Graph, name: String)(m1:Manifest[_],m2:Manifest[_]): Unit
 }
+
+
+abstract class CompactCodeGen extends CompactTraverser {
```

Comment:
refactored this class so that our CodeGen can just inherit the CompactCodeGen, not CompactScalaCodeGen that has Scala Code Gen features